packaging/language/pip.py

```diff
@@ -204,13 +204,36 @@ def _get_full_name(name, version=None):
         resp = name + '==' + version
     return resp
 
+
+def _get_packages(module, pip, chdir):
+    '''Return results of pip command to get packages.'''
+    # Try 'pip list' command first.
+    command = '%s list' % pip
+    rc, out, err = module.run_command(command, cwd=chdir)
+
+    # If there was an error (pip version too old) then use 'pip freeze'.
+    if rc != 0:
+        command = '%s freeze' % pip
+        rc, out, err = module.run_command(command, cwd=chdir)
+
+    return (command, rc, out, err)
+
+
 def _is_present(name, version, installed_pkgs):
     for pkg in installed_pkgs:
-        if '==' not in pkg:
+        # Package listing will be different depending on which pip
+        # command was used ('pip list' vs. 'pip freeze').
```

Comment:
Since ```_get_packages()``` returns the command it used (pip freeze vs pip list) can't we pass that information in here so that we can separately parse freeze output and list output?  That seems less error prone as pip's output changes in the future and other people have to update the conditionals in this function.